manager/integration/tests/test_orphan.py

```diff
@@ -0,0 +1,940 @@
+       in the ready disk
+    """
+
+    lht_hostId = get_self_host_id()
+    node = client.by_id_node(lht_hostId)
+    disks = node.disks
+
+    # test delete disk exception
+    with pytest.raises(Exception) as e:
+        node.diskUpdate(disks={})
+    assert "disable the disk" in str(e.value)
+
+    disk_paths = []
+    # create 1st disks for node
+    node = client.by_id_node(lht_hostId)
+    disks = node.disks
+    path = create_host_disk(client, 'vol-disk-1',
+                            str(Gi), lht_hostId)
+    disk1 = {"path": path, "allowScheduling": True}
+    disk_paths.append(path)
+
+    update_disk = get_update_disks(disks)
+    update_disk["disk1"] = disk1
+    node = node.diskUpdate(disks=update_disk)
+    node = wait_for_disk_update(client, lht_hostId, len(update_disk))
+
+    # create duplicated disks for node
+    node = client.by_id_node(lht_hostId)
+    disks = node.disks
+    duplicated_disk_path = os.path.join(path, 'vol-disk-2')
+    disk_paths.append(duplicated_disk_path)
+    exec_nsenter("mkdir -p {}".format(duplicated_disk_path))
+    disk2 = {"path": duplicated_disk_path, "allowScheduling": True}
+
+    update_disk = get_update_disks(disks)
+    update_disk["disk2"] = disk2
+    node = node.diskUpdate(disks=update_disk)
+    node = wait_for_disk_update(client, lht_hostId, len(update_disk))
+
+    # Create a volume
+    nodes = client.list_node()
+    volume = create_and_check_volume(client, volume_name, len(nodes), SIZE)
+    volume.attach(hostId=lht_hostId, disableFrontend=False)
+    wait_for_volume_healthy(client, volume_name)
+
+    # Create orphaned replica directories
+    orphaned_directories = []
+    orphaned_directories_in_duplicated_disk = []
+    num_orphans = 5
+    for replica in volume.replicas:
+        if replica.hostId != lht_hostId:
+            continue
+        for _ in range(num_orphans):
+            for i, disk_path in enumerate(disk_paths):
+                path = os.path.join(disk_path, "replicas",
+                                    volume_name + "-" + generate_random_id(8))
+                if i == 0:
+                    orphaned_directories.append(path)
+                else:
+                    orphaned_directories_in_duplicated_disk.append(path)
+                exec_nsenter("cp -a {} {}".format(replica.dataPath, path))
+
+    cleanup_volume(client, volume_name)
+
+    # Check orphan CRs
+    count = wait_for_orphan_count(client, num_orphans, 180)
+    assert count == num_orphans
+
+    # Delete orphan CRs
+    orphans = client.list_orphan()
+    for orphan in orphans:
+        delete_orphan(client, orphan.name)
+
+    # wait for 30 sec before finishing the deletion of on-disk data
+    time.sleep(30)
+
+    # The orphaned directories in the ready disk should be deleted
+    for path in orphaned_directories:
+        with pytest.raises(subprocess.CalledProcessError):
+            exec_nsenter("ls {}".format(path))
```

Comment:
Could we use assert instead to verify the number of dirs?

Reply:
`exec_nsenter("ls {}".format(path))` is trying to check if the file exists one by one.
Using assert is more intuitive and undertandable, but `exec_nsenter` can raise the error, so I don't use `assert` here.